### DISCUSSION TITLE
Full screen background image in Next.js using `next/image` & Tailwind CSS without warnings?

### DISCUSSION THREAD

**Discussion by deadcoder0904** [vercel/next.js#55426] [Help]
### Summary

I'm trying to get `next/image` working & it's nothing but a nightmare.

i get random errors every now & then.

today's random error is:

> warn-once.js:16 Image with src "/success.png" has "fill" and a height value of 0. This is likely because the parent element of the image has not been styled to have a set height.

i tried wrapping my `next/image` in a relative parent but i can't get it working.

i also got this:

> Image with src "/success.png" has "fill" and parent element with invalid "position". Provided "static" should be one of absolute,fixed,relative.

my code for the 2nd error works:

```ts
import Link from 'next/link'
import NextImg from 'next/image'

const Success = () => {
  return (
    <main className="isolate min-h-full">
      <NextImg
        src="/success.png"
        alt="success"
        quality={100}
        className="-z-10 object-cover"
        fill
      />
      <div className="mx-auto max-w-7xl px-6 py-32 text-center sm:py-40 lg:px-8">
        <p className="text-base font-semibold leading-8 text-white">
          Success!!!
        </p>
        <h1 className="mt-4 text-3xl font-bold tracking-tight text-white sm:text-5xl">
          Payment Succeeded
        </h1>
        <p className="mt-4 text-base text-white/70 sm:mt-6">
          Congrats, you&apos;ve successfully bought the product!
        </p>
        <div className="mt-10 flex justify-center">
          <Link
            href="/login"
            className="text-sm font-semibold leading-7 text-white"
          >
            <span aria-hidden="true">&larr;</span> Login to Dashboard
          </Link>
        </div>
      </div>
    </main>
  )
}

export default Success
```

but i can't get rid of the console error.

### Additional information

```bash
    Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 20.5.1
      npm: N/A
      Yarn: N/A
      pnpm: 8.7.5
    Relevant Packages:
      next: 13.4.19
      eslint-config-next: 13.4.19
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.2.2
    Next.js Config:
      output: N/A
```

i would love to have a full-screen background image that doesn't throw any errors. ideally, using `next/image`.

i don't mind a tailwind wrapper either. would like to avoid `styles` if i can.

there's literally nothing that is working without errors as most answers are outdated on stackoverflow using next 12. this shouldn't be so hard.

**Comment by vignesh-gupta** [comment]
hey @deadcoder0904 , to use the property `fill` you need to change the position type of parent to `relative` \ `fixed` \ `absolute` 

My recent code- 
```tsx
<div className="w-full h-full rounded-3xl overflow-hidden relative">
  <Image
    src="/hero.jpg"
    fill alt="hero"
    className="w-full h-full object-cover object-center"
  />
</div>
```

and if you want to add a background image just use bg-[url(/hero.jpg)].

Example- 
```tsx
<div className="rounded-3xl bg-[url(/hero.jpg)] md:bg-none overflow-hidden bg-center bg-cover h-full">
    <h1>Some Text</h1>
</div>

```


**NOTE: Look out for the height and width of the div you are giving**

Doc- https://nextjs.org/docs/pages/api-reference/components/image#fill

**Reply by deadcoder0904** [reply]
@vignesh-gupta it doesn't work. i tried the first one.

i don't want to try the 2nd one bcz i want to take advantage of `next/image` property for image optimization.

your 1st code-block literally hides the image & only shows the `div` 3/4th of the way for some reason. 

![image](https://github.com/vercel/next.js/assets/16436270/c0a32165-f7c0-41e3-a73a-00f12bbf092f)

**Reply by ABAGDevelopment** [reply]
I tried to apply background image to a div, However facing lot of challenges , Invalid DOM property `fill-rule`. Did you mean `fillRule`?, Invalid DOM property `clip-rule`. Did you mean `clipRule`?, Invalid DOM property `stroke-linecap`. Did you mean `strokeLinecap`?,Invalid DOM property `stroke-width`. Did you mean `strokeWidth`?, Can anyone help with this
